planner/core/rule_topn_push_down.go

```diff
@@ -96,6 +96,14 @@ func (p *LogicalProjection) pushDownTopN(topN *LogicalTopN) LogicalPlan {
 		for _, by := range topN.ByItems {
 			by.Expr = expression.ColumnSubstitute(by.Expr, p.schema, p.Exprs)
 		}
+
+		// remove meaningless constant sort items.
+		for i := len(topN.ByItems) - 1; i >= 0; i-- {
+			_, isConst := topN.ByItems[i].Expr.(*expression.Constant)
```

Comment:
What if the query is like `select a, b from t order by 2`?

Reply:
`order by 2` is treated as `order by b`, so the execution plan of this statement is:
```sql
TiDB(root@127.0.0.1:test) > desc select a, b from t order by 2 limit 1;
+------------------------+----------+------+------------------------------------------------------------+
| id                     | count    | task | operator info                                              |
+------------------------+----------+------+------------------------------------------------------------+
| TopN_7                 | 1.00     | root | test.t.b:asc, offset:0, count:1                            |
| └─TableReader_15       | 1.00     | root | data:TopN_14                                               |
|   └─TopN_14            | 1.00     | cop  | test.t.b:asc, offset:0, count:1                            |
|     └─TableScan_13     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
+------------------------+----------+------+------------------------------------------------------------+
4 rows in set (0.00 sec)
```